cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/async/UpdateTopicConfigurationRunnable.java

```diff
@@ -9,74 +9,55 @@
-    _goals = parameters.goals();
-    _skipRackAwarenessCheck = parameters.skipRackAwarenessCheck();
-    _dryRun = parameters.dryRun();
-    _modelCompletenessRequirements = parameters.modelCompletenessRequirements();
-    _allowCapacityEstimation = parameters.allowCapacityEstimation();
-    _concurrentInterBrokerPartitionMovements = parameters.concurrentInterBrokerPartitionMovements();
-    _concurrentLeaderMovements = parameters.concurrentLeaderMovements();
-    _skipHardGoalCheck = parameters.skipHardGoalCheck();
-    _replicaMovementStrategy = parameters.replicaMovementStrategy();
-    _replicationThrottle = parameters.replicationThrottle();
-    _excludeRecentlyDemotedBrokers = parameters.excludeRecentlyDemotedBrokers();
-    _excludeRecentlyRemovedBrokers = parameters.excludeRecentlyRemovedBrokers();
     _config = config;
   }
 
   @Override
   protected OptimizationResult getResult() throws Exception {
-    return new OptimizationResult(_kafkaCruiseControl.updateTopicReplicationFactor(_topicPatternByReplicationFactor,
-                                                                                   _goals,
-                                                                                   _skipRackAwarenessCheck,
-                                                                                   _modelCompletenessRequirements,
-                                                                                   _future.operationProgress(),
-                                                                                   _allowCapacityEstimation,
-                                                                                   _concurrentInterBrokerPartitionMovements,
-                                                                                   _concurrentLeaderMovements,
-                                                                                   _skipHardGoalCheck,
-                                                                                   _replicaMovementStrategy,
-                                                                                   _replicationThrottle,
-                                                                                   _excludeRecentlyDemotedBrokers,
-                                                                                   _excludeRecentlyRemovedBrokers,
-                                                                                   _dryRun,
-                                                                                   _uuid),
-                                  _config);
+    if (_topicReplicationFactorChangeParameters != null) {
+      return new OptimizationResult(
+          _kafkaCruiseControl.updateTopicReplicationFactor(_topicReplicationFactorChangeParameters.topicPatternByReplicationFactor(),
+                                                           _topicReplicationFactorChangeParameters.goals(),
```

Comment:
I am wondering whether we can restructure the hierarchy, such that it would be possible to compress `13` parameters in a single `_topicReplicationFactorChangeParameters` parameter passed to `updateTopicReplicationFactor` function?

Reply:
This is a general issue for all the endpoint, I have issue #795 in my to do list, I will address issue as a whole there.